Simplified/Audiobooks/NYPLAudiobookDownloader.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  NYPLAudiobooksDownloader.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2022-04-01.
+//  Copyright © 2022 NYPL. All rights reserved.
+//
+#if FEATURE_AUDIOBOOKS
+import Foundation
+import NYPLAudiobookToolkit
+
+@objc protocol NYPLAudiobookDownloadStatusDelegate {
+  func audiobookDidUpdateDownloadProgress(progress: Float, bookID: String)
+  func audiobookDidCompleteDownload(bookID: String)
+  func audiobookDidCompleteDownloadFirstElement(bookID: String)
+  func audiobookDidReceiveDownloadError(error: NSError?, bookID: String)
+}
+
+class NYPLAudiobookDownloadObject {
+  var bookID: String
+  var audiobookManager: DefaultAudiobookManager
+  
+  init(bookID: String, audiobookManager: DefaultAudiobookManager) {
+    self.bookID = bookID
+    self.audiobookManager = audiobookManager
+  }
+}
+
+@objcMembers class NYPLAudiobookDownloader: NSObject {
```

Comment:
Lemme know what you think, but I'd be in favor of banning using `@objcMembers` if you agree. I find it confusing because it obfuscates what's actually needed by objc vs what's only used by swift.

Reply:
Sure, I will fix it